parser/main.py

```diff
@@ -188,215 +194,125 @@ def parse_request():
+    return parse_and_write_request_bucket("_prod")
 
-    # TODO: add validation for received JSON object
+"""
+Parses incoming request, writes the parsed measurements to InfluxDB bucket (debug or production),
+and sends back parsed measurements back to client.
+"""
+def parse_and_write_request_bucket(bucket):
+    parse_request = flask.request.json
+    message = create_message(parse_request["message"], format_specifier_list)
+    id = message.data.get("ID", "UNKNOWN")
+    type = message.type
 
-    can_msg = StandardFrame(id, data, timestamp, data_length)
+    app.logger.info(f"Received a {message.type} message. ID = {id=}")
 
-    # try extracting measurements from CAN message
+    # try extracting measurements
     try:
-        extracted_measurements: List[Measurement] = can_msg.extract_measurements(CAR_DBC)
-        app.logger.info(f"Successfully parsed CAN message with id={can_msg.hex_identifier}({can_msg.identifier}) and placed into queue")
+        app.logger.info(f"Successfully parsed {type} message with id={id} and placed into queue")
     except Exception:
         app.logger.warn(
-            f"Unable to extract measurements for CAN message with id={can_msg.hex_identifier}({can_msg.identifier})")
+            f"Unable to extract measurements for {type} message with id={id}")
+        app.logger.warn(str(message.data["display_data"]))
         return {
             "result": "PARSE_FAIL",
-            "measurements": [],
-            "id": can_msg.identifier
+            "message": [],
+            "id": id
         }
 
-    # try putting the extracted measurements in the queue for Grafana streaming
-    try:
-        stream_queue.put(extracted_measurements, block=False)
-    except queue.Full:
-        app.logger.warn(
-            "Stream queue full. Unable to add measurements to stream queue!"
-        )
-
     # try writing the measurements extracted
-    for measurement in extracted_measurements:
-        name = measurement.name
-        source = measurement.source
-        m_class = measurement.m_class
-        value = measurement.value
+    for i in range(len(message.data[list(message.data.keys())[0]])):
+        name = message.data["Measurement"][i]
+        source = message.data["Source"][i]
+        m_class = message.data["Class"][i]
+        value = message.data["Value"][i]
+
 
         point = influxdb_client.Point(source).tag("car", CAR_NAME).tag(
             "class", m_class).field(name, value)
-
+        
         # write to InfluxDB
         try:
-            write_api.write(bucket=INFLUX_PROD_BUCKET, org=INFLUX_ORG, record=point)
+            write_api.write(bucket=message.type + bucket, org=INFLUX_ORG, record=point)
             app.logger.info(
-                f"Wrote '{name}' measurement to url={INFLUX_URL}, org={INFLUX_ORG}, bucket={INFLUX_PROD_BUCKET}!")
-        except Exception:
+                f"Wrote '{name}' measurement to url={INFLUX_URL}, org={INFLUX_ORG}, bucket={INFLUX_DEBUG_BUCKET}!")
```

Comment:
`INFLUX_DEBUG_BUCKET` here isn't correct, it should be the `bucket` argument to the function.

Reply:
Corrected!